src/components/implementation/tests/unit_fprr/unit_fprr.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2016, Phani Gadepalli and Gabriel Parmer, GWU, gparmer@gwu.edu.
+ *
+ * This uses a two clause BSD License.
+ */
+
+#include <cos_defkernel_api.h>
+#include <llprint.h>
+#include <res_spec.h>
+#include <sl.h>
+
+/* Ensure this is the same as what is in sl_mod_fprr.c */
+#define SL_FPRR_NPRIOS 32
+
+#define LOWEST_PRIORITY (SL_FPRR_NPRIOS - 1)
+
+#define LOW_PRIORITY (LOWEST_PRIORITY - 1)
+#define HIGH_PRIORITY (LOWEST_PRIORITY - 10)
+
+int lowest_was_scheduled = 0;
+
+void
+low_thread_fn()
+{
+	lowest_was_scheduled = 1;
+	sl_thd_free(sl_thd_curr());
```

Comment:
I feel like `sl_thd_free` should allow to delete the current with just `sl_thd_free(NULL)`. It is not related to the PR but it looks awkward to pass `sl_thd_curr()` to `sl_thd_free()`.

Reply:
@phanikishoreg Honestly, I think another method is in order. `sl_thd_exit()`?